src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -452,12 +452,12 @@ Status OnOffServer::setOnOffValue(chip::EndpointId endpoint, chip::CommandId com
                 ChipLogProgress(Zcl, "ERR: writing on/off %x", to_underlying(status));
                 return status;
             }
+        }
 
-            if (SupportsLightingApplications(endpoint))
-            {
-                ChipLogProgress(Zcl, "Off completed. reset OnTime to  0");
-                Attributes::OnTime::Set(endpoint, 0); // Reset onTime
-            }
+        if (SupportsLightingApplications(endpoint))
+        {
+            ChipLogProgress(Zcl, "Off completed. reset OnTime to  0");
+            Attributes::OnTime::Set(endpoint, 0); // Reset onTime
```

Comment:
Should this happen immediately, or after the level effect bits finish?

@Rob-Houtepen-Signify @jmartinez-silabs

Reply:
@lecndav @bzbarsky-apple I think in the case where the level control fade out to off effect is used, the on time attribute should be resetted once the food out as completed and the onoff state is finally set to false.